aesara/tensor/elemwise.py

```diff
@@ -1288,44 +1280,49 @@ def set_ufunc(self, scalar_op):
     def _output_dtype(self, input_dtype):
         return input_dtype
 
-    def make_node(self, input):
+    def make_node(self, input, *axis):
+        from aesara.graph.basic import Variable
+        from aesara.tensor.basic import as_tensor_variable
+
         input = as_tensor_variable(input)
-        inp_dims = input.type.ndim
         inp_bdcast = input.type.broadcastable
         inp_dtype = input.type.dtype
-        copy_op = False
+        inp_dims = input.type.ndim
 
-        axis = self.axis
-        if axis is None:
-            axis = list(range(len(inp_bdcast)))
+        curr_axis = []
+        for _axis in axis:
+            if isinstance(_axis, Variable):
+                assert isinstance(_axis, TensorConstant)
+                assert (
+                    _axis.ndim == 0
+                ), "Axis values cannot be multi dimensional variables"
+                _axis = int(_axis.data)
+            else:
+                _axis = np.array(_axis, dtype=int)
+                assert _axis.shape == (), "Axis values must be integers"
```

Comment:
These need to be exceptions, because they're dealing more directly with external user/developer input and not just internal code that's intended to operate under very specific assumptions (e.g. `Constant` axes inputs).

Even if these `Op`s are only ever constructed by helper functions that also check these properties, it's still somewhat more reasonable to treat these steps as proper input validation instead of internal assumption checks.